### PR TITLE
Introduce model.Processor

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Introduce type `model.Processor` and field `model.APMEvent.Processor`, which controls the `processor.name` and `processor.event` fields. These fields are used for identifying the event type, used for routing to the appropriate index/data stream and also for monitoring purposes.

Processed event metrics have been moved to a dedicated `model.BatchProcessor`, and out of the model package; they use the Processor field to form the metric name.

Note: Several unused metrics have been removed: stack trace and stack frame counts.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~

## How to test these changes

Enable monitoring, ingest some events, check that `apm-server.processor.<event-type>.transformations` metrics are recorded.

## Related issues

elastic/apm-server#4410
elastic/apm-server#4120
elastic/apm-server#4715

### CODE REVIEW

File: model/modeldecoder/rumv3/decoder.go [hunk 1]
```diff
@@ -161,20 +161,22 @@ func DecodeNestedTransaction(d decoder.Decoder, input *modeldecoder.Input, batch
...
 
 	for _, m := range root.Transaction.Metricsets {
-		metricset := input.Base
-		mapToMetricsetModel(&m, &metricset)
-		metricset.Metricset.Transaction.Name = transaction.Transaction.Name
-		metricset.Metricset.Transaction.Type = transaction.Transaction.Type
-		*batch = append(*batch, metricset)
+		event := input.Base
+		mapToMetricsetModel(&m, &event)
+		event.Processor = model.MetricsetProcessor
```
Comment (stuartnelson3) [inline model/modeldecoder/rumv3/decoder.go:166]: It appears this `event.Processor = model.MetricsetProcessor` is also being set in `mapToMetricsetModel`? Is that intentional?

File: model/modelprocessor/datastream.go [hunk 2]
```diff
@@ -39,14 +39,14 @@ func (s *SetDataStream) ProcessBatch(ctx context.Context, b *model.Batch) error
 }
 
 func (s *SetDataStream) setDataStream(event *model.APMEvent) {
-	switch {
-	case event.Transaction != nil || event.Span != nil:
+	switch event.Processor {
+	case model.SpanProcessor, model.TransactionProcessor:
```
Comment (stuartnelson3) [inline model/modelprocessor/datastream.go:43]: :partying_face:

**Review by stuartnelson3 — APPROVED** [review]
Big fan of the move to explicitly setting the datastream based on `Processor`. One question where it looks like processor might be getting double set, but that's all.

**Review by simitt — APPROVED** [review]

File: model/modeldecoder/rumv3/decoder.go (hunk 1 above)
Comment (axw) [reply model/modeldecoder/rumv3/decoder.go:166]: good catch, I've removed these ones and consolidated on setting it in mapToFooModel

**Pull request merged**